end_to_end/test_llama2_7b.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# The flow of this file is as follows:
+# 1. Download the checkpoint from Meta (https://llama.meta.com/llama-downloads/) in your local directory. Convert this PyTorch checkpoint into Orbax checkpoint format for use in MaxText. 
+# 2. Run decoding, finetuning of Llama2-7b with this converted checkpoint. Also, run pretraining of Llama2-7b.
+# 3. Run decoding from the finetuned weights
+# 4. Convert the scanned checkpoint from step #1 into unscanned checkpoint format and run more efficient decoding. 
+#
+#
+
+set -ex
+idx=$(date +%Y-%m-%d-%H-%M)-${RANDOM}
+
+export M_ENABLE_CHECKPOINTING=true
+# Please point M_BASE_OUTPUT_DIRECTORY to a GCS bucket that you own, this bucket will store all the files generated by MaxText during a run
+export M_BASE_OUTPUT_DIRECTORY=gs://runner-maxtext-logs
+# Please point M_DATASET_PATH to the GCS bucket where you have your training data
+export M_DATASET_PATH=gs://maxtext-dataset
+export M_ASYNC_CHECKPOINTING=false
+
+# We install torch CPU because the checkpoint conversion script MaxText/llama_or_mistral_ckpt.py does not need a TPU/GPU
+pip install torch --index-url https://download.pytorch.org/whl/cpu
+# In the following command, we are copying Meta's checkpoint into a local directory `tmp`. Please update the source `gs://maxtext-llama/llama2-7b/meta-ckpt` to the GCS bucket where you have your Meta checkpoint. 
+# You can use a different local directory than /tmp/, if you do so, please use the same local path for `base-model-path` when running `python3 MaxText/llama_or_mistral_ckpt.py`
+gcloud storage cp -r gs://maxtext-llama/llama2-7b/meta-ckpt /tmp/
+
+# `converted_checkpoint_path` is the path to the GCS bucket where we want to save our converted (Orbax) checkpoint
+export converted_checkpoint_path=gs://maxtext-llama/test/${idx}/decode-ckpt-maxtext
+
+#Next, run the conversion script `MaxText/llama_or_mistral_ckpt.py` to convert Meta's PyTorch checkpoint in `base-model-path` and save the new converted (Orbax) checkpoint in the `maxtext-model-path`
+python3 MaxText/llama_or_mistral_ckpt.py --base-model-path /tmp/meta-ckpt --model-size llama2-7b --maxtext-model-path ${converted_checkpoint_path}
+
+# We define `converted_checkpoint` to refer to the checkpoint subdirectory exactly inside `converted_checkpoint_path`. This way it is easier to use this path in the `train.py` and `decode.py` commands
+export converted_checkpoint=${converted_checkpoint_path}/0/items
+
+# Next, let us run decoding by using this freshly converted checkpoint located at `converted_checkpoint`. Note that this checkpoint only has parameters and no optimizer state. So, we use it by specifying`load_parameters_path=${converted_checkpoint}`
```

Comment:
random nit: "let us" is superfluous in much of your writing! Please remove it!

Reply:
done